specs/plasma.md

```diff
@@ -0,0 +1,114 @@
+# Plasma mode
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [DA Storage](#da-storage)
+- [Input Commitment](#input-commitment)
+- [Data Availability Challenge Contract](#data-availability-challenge-contract)
+- [Derivation](#derivation)
+- [Safety and finality](#safety-and-finality)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
+
+Using Plasma mode means switching to an offchain data availability provider.
+The protocol guarrantees availability with a challenge contract on L1 where users
+can challenge the availability of input data used to derive the chain.
+Challenging an input commitment means forcing providers to submit the input data on chain
+to prove the data is available during a time period. We call the period during which any party
+can challenge a commitment `challenge_window` and the period during which any party
+can submit the input data to resolve the challenge `resolve_window`.
+If a challenge is not resolved by the time `resolve_window` of L1 blocks have passed,
+chain derivation must be reset, omitting the input data when rederiving therefore causing a reorg.
+
+## DA Storage
+
+Input data is uploaded to the storage layer via plain HTTP calls to a stateless DA storage service.
+This service is horizontally scalable and concerned with replicating the data across prefered storage layers
+such as IPFS or any S3 compatible storage. Input data is content addressed by its hash in the request url
+so responses are easily cachable.
+
+Any DA provider can implement the following endpoints to receive and serve input data:
+
+- `POST /put/<hex_encoded_commitment> -H "Content-Type: application/octet-stream"`
+- `GET /get/<hex_encoded_commitment> -H "Content-Type: application/octet-stream"`
+
+## Input Commitment
+
+The batching and compression of input data remain unchanged. When a batch is ready
+to be submitted to the inbox address, the data is uploaded to the DA storage layer instead, and a
+commitment (keccak256 hash) is submitted as the bacher inbox transaction call data.
```

Comment:
Should we mention the type of commitment here or leave it open? (Or describe in more detail below that there can be different types of commitments and how the commitment type determines the `resolve` function's implementation.)

Reply:
It is not consistent to specify the commitment type and then say its flexible?